client/src/components/user-link-metabox/index.js

```diff
@@ -0,0 +1,138 @@
+import React, { useEffect, useState } from 'react';
+import Autocomplete from 'react-autocomplete';
+import PropTypes from 'prop-types';
+import { Button } from '@wordpress/components';
+
+// Hooks.
+import useDebounce from '../../services/use-debounce';
+
+const UserLinkMetaBox = ({
+  linkUserPlaceholder,
+  linkedToLabel,
+  postId,
+  unlinkLabel,
+  user: rawUser,
+  userAlreadyLinked,
+  usersApiUrl,
+}) => {
+  const [search, setSearch] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+  const [user, setUser] = useState({});
+
+  // Debounce search string from input.
+  const debouncedSearchString = useDebounce(search, 750);
+
+  const doUserSearch = (fragment) => {
+    fetch(
+      `${usersApiUrl}?s=${fragment}&post=${postId}`,
+    )
+      .then((res) => res.json())
+      .then((results) => {
+        setSearchResults(results);
+      });
+  };
+
+  const inputProps = {
+    placeholder: linkUserPlaceholder,
+    onKeyDown: (e) => {
+      // If the user hits 'enter', stop the parent form from submitting.
+      if (13 === e.keyCode) {
+        e.preventDefault();
+      }
+    },
+  };
+
+  useEffect(() => {
+    if ('' !== debouncedSearchString) {
+      doUserSearch(debouncedSearchString);
+    }
+  }, [debouncedSearchString]);
+
+  useEffect(() => {
+    setUser(rawUser);
```

Comment:
Could `rawUser` be passed to `useState()`?

Reply:
You almost never want to pass a value directly to state unless you know for sure that it's available on mount. In this case, we could return a null or empty value on mount so should set it in a useEffect hook instead.